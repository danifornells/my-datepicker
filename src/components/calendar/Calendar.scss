/**
 * Component Tokens
 *
 * [NOTE]: 'bourgeois' is the default brand. Used here for syntax highLighting.
 *   On compilation, will be replaced by configured brand.
 */
@import "../../../build/tokens/bourgeois/tokens.calendar";

/** Mixins */
@import "../../styles/mixins/clear-button-styles";

/**
 * Component Styles
 */
.Calendar {
  display: inline-block;
  height: calc(#{$size-spacing-calendar-date-cell} * 8);
  user-select: none;
  width: calc(#{$size-spacing-calendar-date-cell} * 7);

  &-heading {
    align-items: center;
    display: flex;
    height: $size-spacing-calendar-date-cell;
    justify-content: space-between;
  }

  &-navButton {
    @include clear-button-styles;

    fill: $color-text-calendar-nav-button-base;
    height: $size-spacing-calendar-date-cell;
    line-height: 0;
    width: $size-spacing-calendar-date-cell;

    &:hover {
      fill: $color-text-calendar-nav-button-hover;
    }

    &[disabled] {
      cursor: default;
      opacity: 0.25;
      pointer-events: none;
    }
  }

  &-currentMonth {
    font-size: $size-font-calendar-heading;
    font-weight: 600;
  }

  &-page {
    display: grid;
    grid-template-columns: repeat(7, #{$size-spacing-calendar-date-cell});
  }

  &-weekDay {
    display: inline-block;
    font-size: $size-font-calendar-week-day;
    line-height: $size-spacing-calendar-date-cell;
    text-align: center;

    &.is-weekend {
      font-weight: 600;
    }
  }

  &-date {
    @include clear-button-styles;

    font-size: $size-font-calendar-date;
    line-height: $size-spacing-calendar-date-cell;
    transition-duration: 0.25s;
    transition-property: background-color;

    // Apply $shape-calendar-date-base or warns if contains an invalid value

    @if $shape-bounding-box-calendar-date == "circle" {
      border-radius: 50%;
    }

    @else if $shape-bounding-box-calendar-date != "square" {
      @warn "$shape-calendar-date-base: `#{$shape-bounding-box-calendar-date}` is not one of ['circle', 'square']";
    }

    &:hover {
      background-color: $color-background-calendar-date-hover;
      transition-duration: 0.1s;
    }

    &.is-selected {
      background-color: $color-background-calendar-date-selected;
      color: $color-text-calendar-date-selected;
      cursor: default;
      font-weight: 600;
      pointer-events: none;
      transition-duration: 0s;
    }

    &.is-from-another-month,
    &[disabled] {
      cursor: default;
      opacity: 0.25;
      pointer-events: none;
    }
  }
}
