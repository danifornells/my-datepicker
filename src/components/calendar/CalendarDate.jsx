import React from 'react';
import PropTypes from 'prop-types';
import STYLES from './Calendar.scss';

const { bool, func, number } = PropTypes;
const c = (className) => STYLES[className] || 'UNKNOWN';

const CalendarDate = ({
  dateNumber, isFromAnotherMonth, isSelected, isDisabled, onClick,
}) => {
  const classNames = [
    c('Calendar-date'),
    isFromAnotherMonth ? c('is-from-another-month') : '',
    isSelected ? c('is-selected') : '',
  ].join(' ');
  return (
    <button
      className={classNames}
      type="button"
      disabled={isDisabled}
      onClick={onClick}
    >
      {dateNumber}
    </button>
  );
};

CalendarDate.propTypes = {
  dateNumber: number.isRequired,
  isFromAnotherMonth: bool,
  isSelected: bool,
  isDisabled: bool,
  onClick: func,
};

CalendarDate.defaultProps = {
  isFromAnotherMonth: false,
  isSelected: false,
  isDisabled: false,
  onClick: () => {},
};

export default CalendarDate;
