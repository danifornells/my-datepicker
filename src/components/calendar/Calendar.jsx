import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CalendarDate from './CalendarDate';
import IconChevronLeft from '../icons/IconChevronLeft';
import IconChevronRight from '../icons/IconChevronRight';
import * as DATE_LOCALES from '../../locales/date';
import {
  areDatesInSameDay,
  areDatesInSameMonth,
  dateStringToDate,
  dateToDateString,
  getCalendarPageDates,
  isAfter, isBefore,
  isValidDateString,
  isWeekend,
  offsetBy,
  suggestMonthToBeShown,
} from '../../helpers/dates';
import optionalDateString from '../../prop-types/optional-date-string';
import STYLES from './Calendar.scss';

const { string, oneOf, func } = PropTypes;
const c = (className) => STYLES[className] || 'UNKNOWN';

/**
 * [ùóñ] Calendar component
 */
class Calendar extends Component {
  constructor(props) {
    super(props);
    const {
      language, value, min, max,
    } = props;
    this.locales = DATE_LOCALES[language];
    this.minDate = min && isValidDateString(min) ? dateStringToDate(min) : null;
    this.maxDate = max && isValidDateString(max) ? dateStringToDate(max) : null;
    const valueDate = value && isValidDateString(value) ? dateStringToDate(value) : null;
    this.state = {
      currentMonth: suggestMonthToBeShown(valueDate, this.minDate, this.maxDate),
      currentValue: valueDate || null,
    };
  }

  /** [ùóî] When user clicked a new Date */
  handleDayClick(newDate) {
    const { onDateChange } = this.props;
    this.setState({ currentValue: newDate });
    onDateChange(dateToDateString(newDate));
  }

  /** [ùóî] Offsets the calendar by given amount of months */
  offsetByMonth(months) {
    const { currentMonth } = this.state;
    const newCurrentMonth = offsetBy(currentMonth, { months });
    this.setState({ currentMonth: newCurrentMonth });
  }

  /** [ùó•] Months navigation and title (month/year)  */
  renderHeading() {
    const { currentMonth } = this.state;
    const prevMonthDisabled = (
      this.minDate
      && areDatesInSameMonth(currentMonth, this.minDate)
    );
    const nextMonthDisabled = (
      this.maxDate
      && areDatesInSameMonth(currentMonth, this.maxDate)
    );
    const headingText = [
      this.locales.months[currentMonth.getMonth()],
      currentMonth.getFullYear(),
    ].join(' ');
    return (
      <div className={c('Calendar-heading')}>
        <button
          type="button"
          className={c('Calendar-navButton')}
          disabled={prevMonthDisabled}
          onClick={() => this.offsetByMonth(-1)}
        >
          <IconChevronLeft />
        </button>
        <h1 className={c('Calendar-currentMonth')}>{headingText}</h1>
        <button
          type="button"
          className={c('Calendar-navButton')}
          disabled={nextMonthDisabled}
          onClick={() => this.offsetByMonth(1)}
        >
          <IconChevronRight />
        </button>
      </div>
    );
  }

  /** [ùó•] WeekDays names row  */
  renderWeekDays(calendarDates) {
    const { weekdaysShort } = this.locales;
    return Array(7).fill(null).map((d, i) => {
      const sampleDate = calendarDates[i];
      const weekDayName = weekdaysShort[sampleDate.getDay()];
      const isOnWeekend = isWeekend(sampleDate);
      const classNames = [
        c('Calendar-weekDay'),
        isOnWeekend ? c('is-weekend') : '',
      ].join(' ');
      const key = sampleDate.getTime();
      return (
        <span key={key} className={classNames} aria-hidden="true">
          {weekDayName}
        </span>
      );
    });
  }

  /** [ùó•] Date numbers grid  */
  renderDates(calendarDates) {
    const { currentMonth, currentValue } = this.state;
    return calendarDates.map((date) => {
      const key = date.getTime();
      const isFromAnotherMonth = !areDatesInSameMonth(date, currentMonth);
      const isSelected = (currentValue && areDatesInSameDay(date, currentValue));
      const isDisabled = !!((
        isFromAnotherMonth
      ) || (
        this.minDate
        && areDatesInSameMonth(this.minDate, currentMonth)
        && isBefore(date, this.minDate)
      ) || (
        this.maxDate
        && areDatesInSameMonth(this.maxDate, currentMonth)
        && isAfter(date, this.maxDate)
      ));
      const dateNumber = date.getDate();
      return (
        <CalendarDate
          key={key}
          dateNumber={dateNumber}
          isFromAnotherMonth={isFromAnotherMonth}
          isSelected={isSelected}
          isDisabled={isDisabled}
          onClick={() => this.handleDayClick(date)}
        />
      );
    });
  }

  /** [ùó•] Full calendar  */
  render() {
    const { className } = this.props;
    const { currentMonth } = this.state;
    const { firstDay } = this.locales;
    const classNames = [c('Calendar'), className].join(' ');
    const calendarDates = getCalendarPageDates(currentMonth, firstDay);
    return (
      <div className={classNames}>
        {this.renderHeading()}
        <div className={c('Calendar-page')}>
          {this.renderWeekDays(calendarDates)}
          {this.renderDates(calendarDates)}
        </div>
      </div>
    );
  }
}

Calendar.propTypes = {
  /** Min date as DateString */
  min: optionalDateString,
  /** Max date as DateString */
  max: optionalDateString,
  /** Current date as DateString */
  value: optionalDateString,
  /** ClassName/s to be appended */
  className: string,
  /** Language will set locale settings */
  language: oneOf(Object.keys(DATE_LOCALES)),
  /** Invoked when a new date is selected */
  onDateChange: func,
};

Calendar.defaultProps = {
  min: '',
  max: '',
  value: '',
  language: 'en',
  className: '',
  onDateChange: () => {},
};

export default Calendar;
