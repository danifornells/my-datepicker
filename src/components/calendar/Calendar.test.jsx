import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Calendar from './Calendar';
import CalendarDate from './CalendarDate';


describe('Calendar', () => {
  test('should render without throwing an error, and selectable by class "Calendar"', () => {
    expect(shallow(<Calendar />).is('.Calendar')).toBe(true);
  });

  test('should mount in a full DOM', function() {
    expect(mount(<Calendar />).find('.Calendar').length).toBe(1);
  });

  test('should match shallow snapshot', function() {
    expect(shallow(<Calendar />)).toMatchSnapshot();
  });

  test('should match rendered snapshot', function() {
    expect(render(<Calendar />)).toMatchSnapshot();
  });

  test('should have the expected title', function() {
    const renderedCalendar = mount(<Calendar value="2017-10-01" />);
    const monthTitle = renderedCalendar.find('.Calendar-currentMonth');
    expect(monthTitle.text()).toBe('October 2017');
  });

  test('should have working month navigation arrows respecting limits', function() {
    const renderedCalendar = mount(<Calendar
      value="2017-10-01"
      min="2017-09-15"
      max="2017-11-15"
    />);
    const getCurrentElements = () => {
      const monthNavButtons = renderedCalendar.find('.Calendar-navButton');
      return {
        monthNavButtons,
        prevMonth: monthNavButtons.at(0),
        nextMonth: monthNavButtons.at(1),
        monthTitle: renderedCalendar.find('.Calendar-currentMonth')
      }
    };
    // Check current month
    let currentElements = getCurrentElements();
    expect(currentElements.monthNavButtons).toHaveLength(2);
    expect(currentElements.monthTitle.text()).toBe('October 2017');
    expect(currentElements.prevMonth.prop('disabled')).toBe(false);
    expect(currentElements.nextMonth.prop('disabled')).toBe(false);
    // Go to previous month
    currentElements.prevMonth.simulate('click');
    currentElements = getCurrentElements();
    expect(currentElements.monthTitle.text()).toBe('September 2017');
    expect(currentElements.prevMonth.prop('disabled')).toBe(true);
    expect(currentElements.nextMonth.prop('disabled')).toBe(false);
    // Go twice next month
    currentElements.nextMonth.simulate('click');
    currentElements.nextMonth.simulate('click');
    currentElements = getCurrentElements();
    expect(currentElements.monthTitle.text()).toBe('November 2017');
    expect(currentElements.prevMonth.prop('disabled')).toBe(false);
    expect(currentElements.nextMonth.prop('disabled')).toBe(true);
    // Try next month (should be impossible)
    currentElements.nextMonth.simulate('click');
    currentElements = getCurrentElements();
    expect(currentElements.monthTitle.text()).toBe('November 2017');
  });

  test('should have a single expected date selected', function() {
    const renderedCalendar = mount(<Calendar value="2017-10-01"/>);
    const selectedDate = renderedCalendar.find(CalendarDate).filter({ isSelected: true });
    expect(selectedDate).toHaveLength(1);
    expect(selectedDate.prop('dateNumber')).toBe(1);
  });

  test('should enable selecting new date and call onDateChange', function() {
    const onDateChange = jest.fn();
    const renderedCalendar = mount(<Calendar value="2017-10-01" onDateChange={onDateChange}/>);
    // Check dates
    const currentDate = renderedCalendar.find(CalendarDate).at(0);
    const newDate = renderedCalendar.find(CalendarDate).at(1);
    expect(currentDate.prop('isSelected')).toBe(true);
    expect(currentDate.prop('dateNumber')).toBe(1);
    expect(newDate.prop('isSelected')).toBe(false);
    expect(newDate.prop('dateNumber')).toBe(2);
    // Click newDate, and check dates
    newDate.simulate('click');
    const oldCurrentDate = renderedCalendar.find(CalendarDate).at(0);
    const newCurrentDate = renderedCalendar.find(CalendarDate).at(1);
    expect(oldCurrentDate.prop('isSelected')).toBe(false);
    expect(newCurrentDate.prop('isSelected')).toBe(true);
    expect(onDateChange).toHaveBeenCalled();
    expect(onDateChange).toHaveBeenCalledTimes(1);
    expect(onDateChange).toHaveBeenCalledWith('2017-10-02');
  });

});