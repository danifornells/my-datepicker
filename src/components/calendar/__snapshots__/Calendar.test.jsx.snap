// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should match rendered snapshot 1`] = `
<div
  class="Calendar "
>
  <div
    class="Calendar-heading"
  >
    <button
      class="Calendar-navButton"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="Icon "
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.254 16.089c.369.333.398.904.065 1.274a.898.898 0 01-1.27.066l-5.303-4.795a.904.904 0 01.002-1.343l5.256-4.72a.898.898 0 011.271.07c.332.372.3.942-.07 1.275l-4.18 3.753a.4.4 0 000 .594l4.229 3.826z"
        />
      </svg>
    </button>
    <h1
      class="Calendar-currentMonth"
    >
      November 2019
    </h1>
    <button
      class="Calendar-navButton"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="Icon "
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.746 16.089l4.23-3.825a.4.4 0 00-.001-.595l-4.18-3.753a.904.904 0 01-.07-1.275.898.898 0 011.27-.07l5.256 4.72c.399.358.4.983.003 1.342L9.952 17.43a.898.898 0 01-1.271-.066.904.904 0 01.065-1.274z"
        />
      </svg>
    </button>
  </div>
  <div
    class="Calendar-page"
  >
    <span
      aria-hidden="true"
      class="Calendar-weekDay is-weekend"
    >
      Sun
    </span>
    <span
      aria-hidden="true"
      class="Calendar-weekDay "
    >
      Mon
    </span>
    <span
      aria-hidden="true"
      class="Calendar-weekDay "
    >
      Tue
    </span>
    <span
      aria-hidden="true"
      class="Calendar-weekDay "
    >
      Wed
    </span>
    <span
      aria-hidden="true"
      class="Calendar-weekDay "
    >
      Thu
    </span>
    <span
      aria-hidden="true"
      class="Calendar-weekDay "
    >
      Fri
    </span>
    <span
      aria-hidden="true"
      class="Calendar-weekDay is-weekend"
    >
      Sat
    </span>
    <button
      class="Calendar-date is-from-another-month "
      disabled=""
      type="button"
    >
      27
    </button>
    <button
      class="Calendar-date is-from-another-month "
      disabled=""
      type="button"
    >
      28
    </button>
    <button
      class="Calendar-date is-from-another-month "
      disabled=""
      type="button"
    >
      29
    </button>
    <button
      class="Calendar-date is-from-another-month "
      disabled=""
      type="button"
    >
      30
    </button>
    <button
      class="Calendar-date is-from-another-month "
      disabled=""
      type="button"
    >
      31
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      1
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      2
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      3
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      4
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      5
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      6
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      7
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      8
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      9
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      10
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      11
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      12
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      13
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      14
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      15
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      16
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      17
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      18
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      19
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      20
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      21
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      22
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      23
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      24
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      25
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      26
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      27
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      28
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      29
    </button>
    <button
      class="Calendar-date  "
      type="button"
    >
      30
    </button>
  </div>
</div>
`;

exports[`Calendar should match shallow snapshot 1`] = `
<div
  className="Calendar "
>
  <div
    className="Calendar-heading"
  >
    <button
      className="Calendar-navButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <IconChevronLeft
        className=""
        size="medium"
      />
    </button>
    <h1
      className="Calendar-currentMonth"
    >
      November 2019
    </h1>
    <button
      className="Calendar-navButton"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <IconChevronRight
        className=""
        size="medium"
      />
    </button>
  </div>
  <div
    className="Calendar-page"
  >
    <span
      aria-hidden="true"
      className="Calendar-weekDay is-weekend"
      key="Sun"
    >
      Sun
    </span>
    <span
      aria-hidden="true"
      className="Calendar-weekDay "
      key="Mon"
    >
      Mon
    </span>
    <span
      aria-hidden="true"
      className="Calendar-weekDay "
      key="Tue"
    >
      Tue
    </span>
    <span
      aria-hidden="true"
      className="Calendar-weekDay "
      key="Wed"
    >
      Wed
    </span>
    <span
      aria-hidden="true"
      className="Calendar-weekDay "
      key="Thu"
    >
      Thu
    </span>
    <span
      aria-hidden="true"
      className="Calendar-weekDay "
      key="Fri"
    >
      Fri
    </span>
    <span
      aria-hidden="true"
      className="Calendar-weekDay is-weekend"
      key="Sat"
    >
      Sat
    </span>
    <CalendarDate
      dateNumber={27}
      isDisabled={true}
      isFromAnotherMonth={true}
      isSelected={null}
      key="1572127200000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={28}
      isDisabled={true}
      isFromAnotherMonth={true}
      isSelected={null}
      key="1572217200000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={29}
      isDisabled={true}
      isFromAnotherMonth={true}
      isSelected={null}
      key="1572303600000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={30}
      isDisabled={true}
      isFromAnotherMonth={true}
      isSelected={null}
      key="1572390000000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={31}
      isDisabled={true}
      isFromAnotherMonth={true}
      isSelected={null}
      key="1572476400000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={1}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1572562800000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={2}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1572649200000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={3}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1572735600000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={4}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1572822000000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={5}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1572908400000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={6}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1572994800000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={7}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573081200000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={8}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573167600000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={9}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573254000000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={10}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573340400000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={11}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573426800000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={12}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573513200000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={13}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573599600000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={14}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573686000000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={15}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573772400000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={16}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573858800000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={17}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1573945200000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={18}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574031600000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={19}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574118000000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={20}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574204400000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={21}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574290800000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={22}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574377200000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={23}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574463600000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={24}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574550000000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={25}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574636400000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={26}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574722800000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={27}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574809200000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={28}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574895600000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={29}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1574982000000"
      onClick={[Function]}
    />
    <CalendarDate
      dateNumber={30}
      isDisabled={false}
      isFromAnotherMonth={false}
      isSelected={null}
      key="1575068400000"
      onClick={[Function]}
    />
  </div>
</div>
`;
